vyper/codegen/stmt.py

```diff
@@ -93,7 +93,13 @@ def parse_If(self):
     def parse_Log(self):
         event = self.stmt._metadata["type"]
 
-        args = [Expr(arg, self.context).ir_node for arg in self.stmt.value.args]
+        args = []
+        if len(self.stmt.value.keywords) > 0:
+            # keyword arguments
+            args = [Expr(arg.value, self.context).ir_node for arg in self.stmt.value.keywords]
+        else:
+            # positional arguments
+            args = [Expr(arg, self.context).ir_node for arg in self.stmt.value.args]
```

Comment:
factor the code with effects (`Expr()` constructor) to the straight-line code section
```suggestion
        if len(self.stmt.value.keywords) > 0:
            # keyword arguments
            to_compile = [arg.value for arg in self.stmt.value.keywords]
        else:
            # positional arguments
            to_compile = self.stmt.value.args
        args = [Expr(arg, self.context).ir_node for arg in to_compile]
```